File: go/02.go
package main

func fib() int {
	list := []int{1, 2}
	sum := 0

	for {
		inc := list[len(list)-1] + list[len(list)-2]

		if inc > 4000000 {
			break
		}

		if inc % 2 == 0 {
			sum += inc
		}

		list[0] = list[1]
		list[1] = inc
	}

	return sum
}

File: javascript/02.js
function fib() {
  let sum = [1, 2];

  while (true) {
    const inc = sum[sum.length-1] + sum[sum.length-2];
    if (inc > 4000000) break;
    sum.push(inc)
  }

  return sum.reduce((acc , val) => val % 2 == 0 ? acc + val : acc, 0);
}

console.log(fib());
